File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        def fn(root):
            if root == None:
                return 0
            lh = fn(root.left) 
            rh = fn(root.right)
            return 1 + max(lh, rh)
        return fn(root)

File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int ans = 0;
    void fn(TreeNode root, int depth) {
        if (root == null) {
            return;
        }
        this.ans = Math.max(this.ans, depth);
        fn(root.left, depth + 1);
        fn(root.right, depth + 1);
    }
    public int maxDepth(TreeNode root) {
        fn(root, 1);
        return this.ans;
    }
}